packages/react/src/components/UIShell/HeaderPanel.js

```diff
@@ -34,6 +37,15 @@ const HeaderPanel = React.forwardRef(function HeaderPanel(
     [customClassName]: !!customClassName,
   });
 
+  if (__DEV__) {
+    warning(
+      didWarnAboutDeprecation,
+      'The `HeaderPanel` component has been deprecated and will be removed ' +
+        'in the next major release of `carbon-components-react`'
+    );
+    didWarnAboutDeprecation = true;
+  }
+
```

Comment:
Missed just this one `__DEV__`, the others look great 👍 

```suggestion
  warning(
    didWarnAboutDeprecation,
    'The `HeaderPanel` component has been deprecated and will be removed ' +
      'in the next major release of `carbon-components-react`'
  );
  didWarnAboutDeprecation = true;

```